File: cpp/44b_686A_FreeIceCream.cpp
#include<bits/stdc++.h>

int main(){
    unsigned long long n, x, d, cnt = 0;
    char sym;
    std::cin>>n>>x;
    for (int i = 0; i < n; i++){
        std::cin>>sym>>d;
        if(sym == '+'){
            x += d;
        }
        else{
            if(x >= d){
                x -=d;
            }
            else{
                cnt ++;
            }
        }
    }
    std::cout<<x<<" "<<cnt;
    return 0;
}

File: python/44b_686A_FreeIceCream.py
def main():
    cnt = 0
    n, x = input().split()
    n, x = int(n), int(x)
    for _ in range(n):
        sym, d = input().split()
        if sym == "+":
            x += int(d)
        else:
            if x >= int(d):
                x -= int(d)
            else:
                cnt += 1
    print(x, cnt)

if __name__ == "__main__":
    main()